addons/godot_xml/xml.gd

```diff
@@ -150,13 +150,17 @@ static func _parse(xml: PackedByteArray) -> XMLDocument:
 
 static func _make_node(queue: Array, parser: XMLParser):
 	var node_type = parser.get_node_type()
-
+	var node_name = parser.get_node_name()
 	match node_type:
 		XMLParser.NODE_ELEMENT:
 			return _make_node_element(parser)
 		XMLParser.NODE_ELEMENT_END:
+			if queue.is_empty() or queue.back().name != node_name:
+				return
 			return _make_node_element_end(parser)
 		XMLParser.NODE_TEXT:
+			if queue.is_empty():
+				return
```

Comment:
First of all, i can't seem to reproduce it locally. Whatever text i am appending or prepending to the source XMl, Godot properly gives me a NODE_TEXT, so it seems to be an issue on your end, perhaps file a bug on Godot's issue tracker? I think for us the right approach here would be to just strip the input on both ends before parsing. (We need to remove any content before first `<` and any content after the last `>`.) I'll be opening a separate PR that will also (temporarily) ignore any XML after the first root node.

Reply:
Stripping the input is a better idea, but I don’t know how to implement it in GDScript.

Ignoring any XML after the first root node may cause other problems. Perhaps a better approach is to preserve all nodes.

Reply:
> First of all, i can't seem to reproduce it locally. Whatever text i am appending or prepending to the source XMl, Godot properly gives me a NODE_TEXT, so it seems to be an issue on your end, perhaps file a bug on Godot's issue tracker? I think for us the right approach here would be to just strip the input on both ends before parsing. (We need to remove any content before first `<` and any content after the last `>`.) I'll be opening a separate PR that will also (temporarily) ignore any XML after the first root node.

Stripping the input is a better idea, but I don’t know how to implement it in GDScript.

Ignoring any XML after the first root node may cause other problems. Perhaps a better approach is to preserve all nodes.